tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/CallRetry.java

```diff
@@ -50,23 +51,33 @@ public CallRetry(Callable<R> mainAttempt, Callable<Optional<R>> fallbackAttempt)
      * @return Results of main call, or fallback result if data is present after main failure
      */
     public R attemptWithRetry(int maxTries) {
+        log.info("attempt with retry start {}", new Date().toString());
         Throwable lastError = null;
 
         int attemptsMade = 0;
         while (attemptsMade++ < maxTries) {
             try {
-                return mainAttempt.call();
+                R out = mainAttempt.call();
+                log.info("after main attempt {}", new Date().toString());
+                return out;
             } catch (Exception e) {
+                if(exceptionsToNotRetry.stream().anyMatch(exceptionToNotRetry -> e.getClass().equals(exceptionToNotRetry))) {
```

Comment:
`Class.isInstance()` is preferable for handling possible instances of subclasses. e.g. we'd still want to skip retries if a subclass of `SocketException` is caught.
```java
if (exceptionsToNotRetry.stream().anyMatch(c -> c.isInstance(e))) {
    // ...
}
```

Reply:
Updated